doc/tutorials/introduction/test_writing/accuracy_test_writing.markdown

```diff
@@ -0,0 +1,269 @@
+# Writing Accuracy Tests
+
+# {#tutorial_accuracy_test_writing}
+
+`This testing framework is based on Google Tests`
+
+There are two major types of C++ tests: *accuracy/regression* tests and *performance* tests.  Each module can have two test binaries: `opencv_test_<MODULE_NAME>` and `opencv_perf_<MODULE_NAME>`, and two tests folders: `<opencv(_contrib)/modules/<MODULE_NAME>/test>` and `<opencv(_contrib)/modules/<MODULE_NAME>/perf>`. These applications can be built for every supported platform (Win/Lin/Mac/Android).
+
+## ACCURACY TESTS
+
+### Work directory
+
+All modules have their own dir for accuracy tests: `opencv_contrib/modules/(moduleName)/test/...`
+
+### Dir Structure
+
+- `test_precomp.hpp` - file for includes
+- `test_main.cpp` - main file of test sample
+- `test_smth1.cpp` - files for tests
+- `test_smth2.cpp`
+- `...`
+
+### Test structure
+
+```c++
+// name of this case is "name1.name2"
+TEST(name1, name2)
+{
+    ASSERT_....;
+}
+```
+
+### Tests sample example
+
+**Files:**
+
+- test_precomp.hpp
+- test_main.cpp
+- test_sum.cpp
+- test_sub.cpp
+
+
+
+```c++
+// test_precomp.hpp
+
+#ifndef __OPENCV_TEST_PRECOMP_HPP__
+#define __OPENCV_TEST_PRECOMP_HPP__
+
+#include <...>
+#include "..."
+
+namespace opencv_test {
+  using namespace cv::(moduleName);
+}
+
+#endif
+```
+
+
+
+```c++
+// test_main.cpp
+#include "test_precomp.hpp"
+
+CV_TEST_MAIN("<FOLDER_NAME_IN_TESTDATA>")
+```
+
+
+
+```c++
+// test_sum.cpp
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+TEST(SUM, int)
+{
+    int a=1;
+    int b=2;
+    int res=a+b;
+    ASSERT_EQ(res, 3)
+}
+
+TEST(SUM, float)
+{
+    float a=0.1f;
+    float b=0.2f;
+    float res=a+b;
+    ASSERT_EQ(res, 0.3f)
+}
+
+}} // namespace
+```
+
+
+
+
+```c++
+// test_sub.cpp
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+TEST(SUB, int)
+{
+    int a=2;
+    int b=1;
+    int res=a-b;
+    ASSERT_EQ(res, 1)
+}
+
+TEST(SUB, float)
+{
+    float a=0.2f;
+    float b=0.1f;
+    float res=a-b;
+    ASSERT_EQ(res, 0.1f)
+}
+
+}} // namespace
+```
```

Comment:
These samples says nothing about best practices from these areas:
- test data handling
- images loading / processing
- comparing results (hopefully I will not see `countNonZero()` again)

For code above it is enough put the link of Google Tests documentation (it contains much more useful information)

Reply:
link added